`wails dev` panics due to application immediately shutting down

### Description

I'm using wails on an externally built project. In any case, a separate command line builds the files (esbuild and tailwinds watch are used), and they are dumped in `frontend/src`. 

From what I can tell, the application seems to invoke shutdown instantly and then crashes down the callstack (maybe because of the empty `frontend:x` fields in the config?).

This is the result of running `wails dev`:

```
C:\Users\admin\Documents\GitHub\hitta\hittawails>wails dev
Wails CLI v2.0.0-beta.42

Executing: go mod tidy -compat=1.17
Executing: wails generate module
Building application for development...
  - No Install command. Skipping.
  - No Build command. Skipping.
  - Generating bundle assets: Done.
  - Compiling application: Done.

Watching (sub)/directory: C:\Users\admin\Documents\GitHub\hitta\hittawails
Using DevServer URL: http://localhost:34115
Using Frontend DevServer URL: auto
Using reload debounce setting of 100 milliseconds
INF | Serving assets from frontend DevServer URL: auto
DEB | WebView2 Runtime Version '103.0.1264.77' installed. Minimum version required: 94.0.992.31.
Added new directory to watcher: C:\Users\admin\Documents\GitHub\hitta\hittawails\frontend\wailsjs\go
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x6bede6]

goroutine 1 [running, locked to thread]:
github.com/wailsapp/wails/v2/internal/frontend/desktop/windows.(*Frontend).Quit(0xc000125f08)
        C:/Users/admin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.42/internal/frontend/desktop/windows/frontend.go:375 +0x46
github.com/wailsapp/wails/v2/internal/frontend/devserver.(*DevWebServer).Quit(0x8bb0b0)
        C:/Users/admin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.42/internal/frontend/devserver/devserver.go:152 +0x22
github.com/wailsapp/wails/v2/internal/appng.(*App).Shutdown(0xc00019e990)
        C:/Users/admin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.42/internal/appng/app_dev.go:51 +0x44
github.com/wailsapp/wails/v2/internal/appng.(*App).Run(0xc000074460)
        C:/Users/admin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.42/internal/appng/app_dev.go:56 +0x45
github.com/wailsapp/wails/v2.Run(0xc0001be100)
        C:/Users/admin/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.42/wails.go:40 +0xf4
main.main()
        C:/Users/admin/Documents/GitHub/hitta/hittawails/main.go:19 +0x1fb
```

### To Reproduce

My wails config is as follows. I have empty strings for install, build and run as these are handled externally:

```
{
  "name": "hitta",
  "outputfilename": "hittawails",
  "frontend:install": "",
  "frontend:build": "",
  "frontend:dev:watcher": "",
  "frontend:dev:serverUrl": "auto",
  "author": {
    "name": "Max Kunes",
    "email": "myemail"
  }
}
```

The only modification I have made to the actual generated go project is embedding frontend/src instead of frontend:

```
//go:embed frontend/src
var assets embed.FS
```



### Expected behaviour

I expect that wails should correctly display my `index.html` without the use of `frontend:X` commands.

### Screenshots

Frontend Directory:
![explorer_5saEB7mNpJ](https://user-images.githubusercontent.com/11730836/183277570-d4b04c16-3687-4865-a5f1-5e7f7c1dea46.png)

Frontend/src Directory:

![image](https://user-images.githubusercontent.com/11730836/183277573-a25f58d4-489e-4e03-bfa3-5a788a6abd5d.png)


### Attempted Fixes

_No response_

### System Details

```shell
PS C:\Users\admin\Documents\GitHub\hitta\hittawails\frontend\src> wails doctor
Wails CLI v2.0.0-beta.42

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:             Windows 10 Pro
Version:        2009 (Build: 22000)
ID:             21H2
Go Version:     go1.17.6
Platform:       windows
Architecture:   amd64

Wails
------
Version:        v2.0.0-beta.42

Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       103.0.1264.77
npm             N/A             Installed       8.1.2
*upx            N/A             Available
*nsis           N/A             Available

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details:
  - upx : Available at https://upx.github.io/
  - nsis : Available at https://nsis.sourceforge.io/Download



If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

PS C:\Users\admin\Documents\GitHub\hitta\hittawails\frontend\src>
```
```


### Additional context

_No response_

--- a/v2/cmd/wails/internal/commands/dev/dev.go
+++ b/v2/cmd/wails/internal/commands/dev/dev.go
@@ -198,8 +198,9 @@ func AddSubcommand(app *clir.Cli, w io.Writer) error {
 		}
 
 		// frontend:dev:watcher command.
+		frontendDevAutoDiscovery := projectConfig.IsFrontendDevServerURLAutoDiscovery()
 		if command := projectConfig.DevWatcherCommand; command != "" {
-			closer, devServerURL, err := runFrontendDevWatcherCommand(cwd, command, projectConfig.FrontendDevServerURL == "auto")
+			closer, devServerURL, err := runFrontendDevWatcherCommand(cwd, command, frontendDevAutoDiscovery)
 			if err != nil {
 				return err
 			}
@@ -208,6 +209,8 @@ func AddSubcommand(app *clir.Cli, w io.Writer) error {
 				flags.frontendDevServerURL = devServerURL
 			}
 			defer closer()
+		} else if frontendDevAutoDiscovery {
+			return fmt.Errorf("Unable to auto discover frontend:dev:serverUrl without a frontend:dev:watcher command, please either set frontend:dev:watcher or remove the auto discovery from frontend:dev:serverUrl")
 		}
 
 		// Do initial build but only for the application.

--- a/v2/internal/project/project.go
+++ b/v2/internal/project/project.go
@@ -103,6 +103,10 @@ func (p *Project) GetDevInstallerCommand() string {
 	return p.InstallCommand
 }
 
+func (p *Project) IsFrontendDevServerURLAutoDiscovery() bool {
+	return p.FrontendDevServerURL == "auto"
+}
+
 func (p *Project) Save() error {
 	data, err := json.MarshalIndent(p, "", "  ")
 	if err != nil {

